engine/access/rest/rest_api_server.go

```diff
@@ -0,0 +1,152 @@
+package rest
+
+import (
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/engine/access/rest/generated"
+)
+
+// NewRestAPIServer returns an HTTP server initialized with the REST API handler
+func NewRestAPIServer(api *APIHandler, listenAddress string, logger zerolog.Logger) *http.Server {
+
+	router := mux.NewRouter().StrictSlash(true)
+	for _, route := range apiRoutes(api) {
+		var handler http.Handler
+		handler = route.HandlerFunc
+		handler = newHandler(handler, route.Name, logger)
+		router.
+			Methods(route.Method).
+			Path(route.Pattern).
+			Name(route.Name).
+			Handler(handler)
+	}
+
+	return &http.Server{
+		Addr:    listenAddress,
```

Comment:
A good practice to set timeouts to avoid Slowloris attacks.
```
        WriteTimeout: time.Second * 15,
        ReadTimeout:  time.Second * 15,
        IdleTimeout:  time.Second * 60,
```
Potentially even expose these args and make them configurable but maybe not necessary.